pkg/test/echo/docker/Dockerfile.app_sidecar

```diff
@@ -1,6 +1,33 @@
-ARG BASE_VERSION=latest
+ARG VM_IMAGE_NAME
+ARG VM_IMAGE_VERSION
+FROM ${VM_IMAGE_NAME}:${VM_IMAGE_VERSION}
+# Dockerfile for different VM OS versions
+ENV DEBIAN_FRONTEND=noninteractive
 
-FROM docker.io/istio/base:${BASE_VERSION}
+# Do not add more stuff to this list that isn't small or critically useful.
+# If you occasionally need something on the container do
+# sudo apt-get update && apt-get whichever
+
+# hadolint ignore=DL3005,DL3008
+RUN apt-get update && \
+    apt-get install --no-install-recommends -y \
+      iptables \
+      iproute2 \
+      sudo \
+   && apt-get upgrade -y \
+   && apt-get clean \
+   && rm -rf  /var/log/*log /var/lib/apt/lists/* /var/log/apt/* /var/lib/dpkg/*-old /var/cache/debconf/*-old
+
+# Fix the bug of --to-ports not available. Redeclare ARGs to avoid reset
+ARG VM_IMAGE_NAME
```

Comment:
how is it reset? from FROM?

Reply:
During my experiments, the variable will be reset after each phase of the build. I would need to declare it again, according to https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact

Reply:
Updated the wording. It's not "reset". It's just out of scope.